File: test/fixtures/detectors/empty-error-handler.ts
// === Should be flagged ===

// Empty catch block
try {
  riskyOperation();
} catch (e) {}

// Console.log-only catch block
try {
  riskyOperation();
} catch (e) {
  console.log(e);
}

// Console.error-only catch block
try {
  riskyOperation();
} catch (e) {
  console.error(e);
}

// Console.warn-only catch block
try {
  riskyOperation();
} catch (e) {
  console.warn(e);
}

// === Should NOT be flagged ===

// Has a comment — intentionally empty
try {
  riskyOperation();
} catch (e) {
  /* intentionally empty */
}

// Re-throws the error
try {
  riskyOperation();
} catch (e) {
  throw new Error("Failed");
}

// Has recovery logic (return)
try {
  riskyOperation();
} catch (e) {
  return fallbackValue;
}

// Multiple statements including re-throw
try {
  riskyOperation();
} catch (e) {
  cleanup();
  throw e;
}

// Has line comment — intentionally empty
try {
  riskyOperation();
} catch (e) {
  // ignore this error
}

function riskyOperation() {}
function cleanup() {}
const fallbackValue = null;

File: test/fixtures/detectors/empty-error-handler.py
# === Should be flagged ===

# except with only pass
try:
    risky_operation()
except:
    pass

# except Exception with only pass
try:
    risky_operation()
except Exception as e:
    pass

# except with only print
try:
    risky_operation()
except Exception as e:
    print(e)

# === Should NOT be flagged ===

# except with comment and pass
try:
    risky_operation()
except Exception as e:
    # intentionally empty
    pass

# Re-raises the error
try:
    risky_operation()
except Exception as e:
    raise ValueError("failed")

# Has recovery logic
try:
    risky_operation()
except Exception as e:
    return fallback_value

# Multiple statements
try:
    risky_operation()
except Exception as e:
    cleanup()
    raise


def risky_operation():
    pass

def cleanup():
    pass

fallback_value = None
